importer/chunk/splitting.go

```diff
@@ -12,7 +13,10 @@ var log = logging.Logger("chunk")
 var DefaultBlockSize int64 = 1024 * 256
 
 type Splitter interface {
-	NextBytes() ([]byte, error)
+	// returns the data, an offset if applicable and, an error condition
+	NextBytes() ([]byte, int64, error)
```

Comment:
I think instead of adding an offset here to the return (which will pretty much only be used by this codepath). We should change the `[]byte` return to be some type that allows us to either return data, or to return a file piece, similar to the change made to the block object. This new type (whatever we end up making) can also replace the data field in the merkledag.Node object to simplify the `DataPtr` stuff you've got going.